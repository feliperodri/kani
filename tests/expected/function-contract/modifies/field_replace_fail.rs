// Copyright Kani Contributors
// SPDX-License-Identifier: Apache-2.0 OR MIT
// kani-flags: -Zfunction-contracts

struct S<'a> {
    distraction: usize,
    target: &'a mut u32,
}
#[kani::requires(*s.target < 100)]
#[kani::modifies(s.target)]
fn modify(s: S) {
    *s.target += 1;
}

#[kani::proof]
#[kani::stub_verified(modify)]
fn main() {
    let mut i = kani::any_where(|i| *i < 100);
    let i_copy = i;
    let s = S { distraction: 0, target: &mut i };
    modify(s);
    kani::assert(i == i_copy + 1, "Increment havocked");
}
