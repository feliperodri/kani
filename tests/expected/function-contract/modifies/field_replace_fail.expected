main.assertion\
- Status: FAILURE\
- Description: "Increment havocked"

Failed Checks: Increment havocked

VERIFICATION:- FAILED
